src/App.tsx

```diff
@@ -14,50 +16,95 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  SORT_FIELD_ALPHABET = 'alphabet',
+  SORT_FIELD_BY_LENGTH = 'length',
+  SORT_FIELD_RESET = '',
+}
+
+function getPreparedGoods(goods:string[], sortField:SortType, reverse:boolean) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.SORT_FIELD_BY_LENGTH:
+          return good1[sortField] - good2[sortField];
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField]
+  = useState<SortType>(SortType.SORT_FIELD_RESET);
+  const [reverse, setReverse] = useState(false);
+  const readyGoods = getPreparedGoods(goodsFromServer, sortField, reverse);
 
-export const App: React.FC = () => {
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SortType.SORT_FIELD_ALPHABET);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.SORT_FIELD_BY_LENGTH,
+          })}
+          onClick={() => {
+            setSortField(SortType.SORT_FIELD_BY_LENGTH);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => {
+            setReverse(!reverse);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField !== '' || reverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
```

Comment:
```suggestion
            onClick={reset}
```
Consider moving this logic to a separate function from jsx